cloud/pkg/cloudhub/common/model/types.go

```diff
@@ -72,14 +72,15 @@ type HubInfo struct {
 
 // NewResource constructs a resource field using resource type and ID
 func NewResource(resType, resID string, info *HubInfo) string {
-	var prefix string
+	paths := make([]string, 4)
 	if info != nil {
-		prefix = fmt.Sprintf("%s/%s/", model.ResourceTypeNode, info.NodeID)
+		paths = append(paths, model.ResourceTypeNode, info.NodeID)
 	}
-	if resID == "" {
-		return fmt.Sprintf("%s%s", prefix, resType)
+	paths = append(paths, resType)
+	if resID != "" {
+		paths = append(paths, resID)
 	}
-	return fmt.Sprintf("%s%s/%s", prefix, resType, resID)
+	return path.Join(paths...)
```

Comment:
Although this may not cause any problems，it is worth noting that this PR will cause the result of the function to change.

When `NewResource("", "id",nil)` is called, the return value will change from `/id` to `id`.

Regardless of whether `resType` can be empty, the format should be `resType/resID` by semantics, even if the type is empty, the `/` should be preserved.

@fisherxu What do you think?

Reply:
```golang
var prefix string
if info != nil {
    prefix = fmt.Sprintf("%s/%s/", model.ResourceTypeNode, info.NodeID)
}
if resID == "" {
    return fmt.Sprintf("%s%s", prefix, resType)
}
return fmt.Sprintf("%s%s/%s", prefix, resType, resID)
```
By the way, in the historical version of the code , when ‘resID == “”’, the constructor return value will be:
  1.ResourceTypeNode/info.NodeID/resType
  2. resType
  3. ""
Waiting for the outcome of the discussion, thx.

Reply:
Make sense, I'd prefer not modify this, have not seen significant improvement, and also not quite sure about the impact.

Reply:
Ok~